### ISSUE TITLE
Alan Wake 2 crashes at startup due to dxil-spirv assert exception

### ISSUE THREAD

**Issue by lucasomigli** [HansKristian-Work/vkd3d-proton#1788]
Hello

### Software information
 Alan Wake II
env variables:
VKD3D_CONFIG=dxr
GAMEMODERUN=1
VKD3D_DEBUG=trace
VKD3D_FEATURE_LEVEL=12_2
VKD3D_SHADER_DEBUG=trace

### System information
- GPU: AMD Radeon RX 6600 XT
- Driver: Mesa 24.0~git2311200600.e3afbd~oibaf~j
- Wine version: lutris-GE-Proton-8-22-x86_64
- VKD3D-Proton version: 2.8

### Log files
[vkd3d_luca.log](https://github.com/HansKristian-Work/vkd3d-proton/files/13420264/vkd3d_luca.log)

Just trying to lunch this via Lutris and looking at the logs I see the same error even after trying different vd3d versions.

`0Assertion failed: b->immediate_post_dominator, file ../../../home/maister/git/vkd3d-proton/subprojects/dxil-spirv/node.cpp, line 295`

I did eventually look into one of the fxil-spirv releases to see where the assertion is failing but not clear what this specifically indicates in line 295 of [home/maister/git/vkd3d-proton/subprojects/dxil-spirv/node.cpp
](https://github.com/HansKristian-Work/dxil-spirv/blob/master/node.cpp)

**Comment by Blisto91** [comment]
> I see the same error even after trying different vd3d versions.

Does that include vkd3d-proton 2.10?

**Comment by lucasomigli** [comment]
> Does that include vkd3d-proton 2.10?

After further checking, with vkd3d-proton 2.10 it turns out the lutris runtime was throwing: `
err:vkd3d-proton:load_d3d12core_once: Failed to find vkd3d-proton d3d12core interfaces. Make sure d3d12core.dll is installed as well.
`

Found out the reason for that. The Lutris runtime library is missing a mention to fetch d3d12core at runtime causing vkd3d library to fail.
This is an issue with the lutris runtime hence closing this (not an issue with vkd3d) as mentioned here: https://github.com/lutris/lutris/issues/4782

To recap, if you have this, make sure to use the latest vkd3d-proton version (as of now 2.10) and manually edit the file in the lutris runtime folder`lutris/util/wine/vkd3d.py` to add for the missing dll.

**Comment by lucasomigli** [comment]
> @lucasomigli It should be noted that the current version of Lutris is 0.5.14 and Lutris does _not_ support anything else except for its very latest release. In other words, anything older than 0.5.14 is considered unsupported / best-effort / if it breaks, you get to keep the pieces, and any parts of its shipped runtime are only tested against the latest release. You may want to consider upgrading Lutris to avoid surprises.

Sorry yes, thanks for pointing that out. I was using an apparently defunct PPA which does not have the latest version. Confirmed I upgraded to the latest which does fix the bug.
